packages/sitecore-jss/src/personalize/cdp-service.ts

```diff
@@ -85,18 +93,32 @@ export class CdpService {
     debug.personalize('executing experience for %s %s %o', contentId, browserId, context);
 
     const fetcher = this.config.dataFetcherResolver
-      ? this.config.dataFetcherResolver<ExecuteExperienceResult>()
+      ? this.config.dataFetcherResolver<ExecuteExperienceResult>({ timeout: this.timeout })
       : this.getDefaultFetcher<ExecuteExperienceResult>();
-    const response = await fetcher(endpoint, {
-      clientKey: this.config.clientKey,
-      pointOfSale: this.config.pointOfSale,
-      browserId,
-      context,
-    });
-    return {
-      variantId: response.data.variantId || undefined, // coerce empty strings to undefined
-      browserId: response.data.browserId || undefined,
-    };
+
+    try {
+      const response = await fetcher(endpoint, {
+        clientKey: this.config.clientKey,
+        pointOfSale: this.config.pointOfSale,
+        browserId,
+        context,
+      });
+      response.data.variantId === '' && (response.data.variantId = undefined);
+      return response.data;
+    } catch (error) {
+      if (
+        (error as AxiosError).code === '408' ||
+        (error as ResponseError).response?.status === 408 ||
+        (error as Error).name === 'AbortError' ||
+        /timeout/i.test((error as Error).message)
```

Comment:
Since we can use any kind of fetcher, this makes sure we catch any timeout event(as last resort, assuming the message includes the 'timeout'). If we remove that we will need to check for specific messages like "timeout of 30ms exceeded"( we get this in our tests) where the ms might not be a specific value or maybe the message itself is formatted in a different way. About the cases where we weren't the initiator since this runs only in"executeExperience" the timeout should come from this call and we should return the fallback values. Of course, any suggestions are more than welcome.

Reply:
I guess sniffing for text in the error message seems like it shouldn't be necessary (we should be able to rely on status code or error name). Are you saying this is actually a case you encountered and is necessary (i.e. the message  was the only indicator of abort)?